logger_test.go

```diff
@@ -0,0 +1,63 @@
+package analytics
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"log"
+	"testing"
+)
+
+type testLogger struct {
+	logs   string
+	errors string
+}
+
+func (t *testLogger) Logf(format string, args ...interface{}) {
+	t.logs += fmt.Sprintf("INFO "+format+"\n", args...)
+}
+
+func (t *testLogger) Errorf(format string, args ...interface{}) {
+	t.errors += fmt.Sprintf("ERROR "+format+"\n", args...)
+}
+
+// This test ensures that the interface doesn't get changed and stays compatible
```

Comment:
could we just do `var _ Logger = (*testing.T)(nil)` to assert interface compliance?